app/embedder.py

```diff
@@ -0,0 +1,117 @@
+import logging
+from bert import tokenization as token
+from datetime import datetime
+from ming import Field
+from ming import schema
+from ming import Session
+from ming import create_datastore
+from ming.declarative import Document
+import numpy
+import tensorflow as tf
+import tensorflow_hub as hub
+
+# Used for making sure all sentences end up
+# padded to equivalent vector lengths.
+MAX_SEQ_LENGTH = 256
+
+bind = create_datastore('tutorial')
+session = Session(bind)
+
+
+class Embedding(Document):
+    """Python representation of embedding cache schema in MongoDB."""
+
+    class __mongometa__:
+        session = session
+        name = 'embedding_matrix'
+
+    _id = Field(schema.ObjectId)
+    bert = Field(schema.String)
+    seq = Field(schema.String)
+    embedding = Field(schema.Array(schema.Array(schema.Float)))
+    update_timestamp = Field(datetime, if_missing=datetime.utcnow)
+
+
+class Embedder:
+    def __init__(self, bert: str):
+        self.bert = bert
+        self.tf_hub = hub.Module(bert, trainable=True)
+
+        t_info = self.tf_hub(signature="tokenization_info", as_dict=True)
+        with tf.compat.v1.Session() as sess:
+            # vocab_file is a BERT-global, stable mapping {tokens: ids}
+            vocab_file, do_lower_case = sess.run(
+                [t_info["vocab_file"],
+                 t_info["do_lower_case"]])
+            self.tokenizer = token.FullTokenizer(
+                vocab_file=vocab_file,
+                do_lower_case=do_lower_case)
+
+    def GetEmbedding(self, seq: str):
+        # fetch from cache
+        obj = Embedding.m.get(bert=self.bert, seq=seq)
+        if obj:
+            logging.info("Using embedding matrix fetched from MongoDB...")
+            # convert list back to numpyArray
+            matrix = numpy.array(obj.embedding)
+            logging.debug(matrix)
+            return matrix
+
+        logging.info("Embedding matrix for bert=%s, seq=%s "
+                     "not found in cache. Generating..." %
+                     (self.bert, seq))
+        matrix = self.buildEmbedding(seq)
+
+        # convert numpyArray to list for storage in MongoDB
+        l_matrix = matrix.tolist()
+        e = Embedding.make(
+            dict(bert=self.bert, seq=seq, embedding=l_matrix))
+        e.m.save()
+        logging.info("Embedding matrix stored in MongoDB.")
+        return matrix
+
+    def buildEmbedding(self, seq: str):
```

Comment:
Did you mean to make this private (as in _buildEmbedding)?

Reply:
I did. Apparently been writing too much Go lately...